### PR TITLE
Use checkers on Integration test when possible

### PR DESCRIPTION
**Pull request by HuKeping**
A part of #16756

Signed-off-by: Hu Keping hukeping@huawei.com

### CODE REVIEW

**Comment by HuKeping** [comment]
The windows failure seems unrelated.

File: integration-cli/docker_cli_tag_test.go [hunk 1]
```diff
@@ -27,14 +28,11 @@ func (s *DockerSuite) TestTagUnprefixedRepoByID(c *check.C) {
...
 func (s *DockerSuite) TestTagInvalidUnprefixedRepo(c *check.C) {
-
 	invalidRepos := []string{"fo$z$", "Foo@3cc", "Foo$3", "Foo*3", "Fo^3", "Foo!3", "F)xcz(", "fo%asd"}
 
 	for _, repo := range invalidRepos {
 		_, _, err := dockerCmdWithError("tag", "busybox", repo)
-		if err == nil {
-			c.Fatalf("tag busybox %v should have failed", repo)
-		}
+		c.Assert(err, check.NotNil, check.Commentf("tag busybox %v should have failed", repo))
```
Comment (cpuguy83) [inline integration-cli/docker_cli_tag_test.go]: Seeing the output from the command here would be great.

File: integration-cli/docker_cli_tag_test.go [hunk 2]
```diff
@@ -46,9 +44,7 @@ func (s *DockerSuite) TestTagInvalidPrefixedRepo(c *check.C) {
 
 	for _, repotag := range invalidTags {
 		_, _, err := dockerCmdWithError("tag", "busybox", repotag)
-		if err == nil {
-			c.Fatalf("tag busybox %v should have failed", repotag)
-		}
+		c.Assert(err, check.NotNil, check.Commentf("tag busybox %v should have failed", repotag))
```
Comment (cpuguy83) [inline integration-cli/docker_cli_tag_test.go]: Seeing the output from the command here would be great.

File: integration-cli/docker_cli_tag_test.go [hunk 3]
```diff
@@ -80,9 +76,9 @@ func (s *DockerSuite) TestTagExistedNameWithoutForce(c *check.C) {
 
 	dockerCmd(c, "tag", "busybox:latest", "busybox:test")
 	out, _, err := dockerCmdWithError("tag", "busybox:latest", "busybox:test")
-	if err == nil || !strings.Contains(out, "Conflict: Tag busybox:test is already set to image") {
-		c.Fatal("tag busybox busybox:test should have failed,because busybox:test is existed")
-	}
+
+	c.Assert(err, check.NotNil)
```
Comment (cpuguy83) [inline integration-cli/docker_cli_tag_test.go]: Seeing the output from the command here would be great.

File: integration-cli/docker_cli_tag_test.go [hunk 4]
```diff
@@ -101,21 +97,21 @@ func (s *DockerSuite) TestTagWithPrefixHyphen(c *check.C) {
 	if err := pullImageIfNotExist("busybox:latest"); err != nil {
 		c.Fatal("couldn't find the busybox:latest image locally and failed to pull it")
 	}
+
 	// test repository name begin with '-'
 	out, _, err := dockerCmdWithError("tag", "busybox:latest", "-busybox:test")
-	if err == nil || !strings.Contains(out, "repository name component must match") {
-		c.Fatal("tag a name begin with '-' should failed")
-	}
+	c.Assert(err, check.NotNil)
```
Comment (cpuguy83) [inline integration-cli/docker_cli_tag_test.go]: Seeing the output from the command here would be great.

File: integration-cli/docker_cli_tag_test.go [hunk 5]
```diff
@@ -101,21 +97,21 @@ func (s *DockerSuite) TestTagWithPrefixHyphen(c *check.C) {
...
+	c.Assert(out, checker.Contains, "repository name component must match", check.Commentf("tag a name begin with '-' should failed"))
+
 	// test namespace name begin with '-'
 	out, _, err = dockerCmdWithError("tag", "busybox:latest", "-test/busybox:test")
-	if err == nil || !strings.Contains(out, "repository name component must match") {
-		c.Fatal("tag a name begin with '-' should failed")
-	}
+	c.Assert(err, check.NotNil)
```
Comment (cpuguy83) [inline integration-cli/docker_cli_tag_test.go]: Seeing the output from the command here would be great.

File: integration-cli/docker_cli_tag_test.go [hunk 6]
```diff
@@ -101,21 +97,21 @@ func (s *DockerSuite) TestTagWithPrefixHyphen(c *check.C) {
...
+	c.Assert(out, checker.Contains, "repository name component must match", check.Commentf("tag a name begin with '-' should failed"))
+
 	// test index name begin with '-'
 	out, _, err = dockerCmdWithError("tag", "busybox:latest", "-index:5000/busybox:test")
-	if err == nil || !strings.Contains(out, "Invalid index name (-index:5000). Cannot begin or end with a hyphen") {
-		c.Fatal("tag a name begin with '-' should failed")
-	}
+	c.Assert(err, check.NotNil)
```
Comment (cpuguy83) [inline integration-cli/docker_cli_tag_test.go]: Seeing the output from the command here would be great.

**Comment by HuKeping** [comment]
@cpuguy83 updated.

File: integration-cli/docker_cli_tag_test.go [hunk 7]
```diff
@@ -27,14 +28,11 @@ func (s *DockerSuite) TestTagUnprefixedRepoByID(c *check.C) {
...
-
 	invalidRepos := []string{"fo$z$", "Foo@3cc", "Foo$3", "Foo*3", "Fo^3", "Foo!3", "F)xcz(", "fo%asd"}
 
 	for _, repo := range invalidRepos {
-		_, _, err := dockerCmdWithError("tag", "busybox", repo)
-		if err == nil {
-			c.Fatalf("tag busybox %v should have failed", repo)
-		}
+		out, _, err := dockerCmdWithError("tag", "busybox", repo)
+		c.Assert(err, check.NotNil, check.Commentf(out))
```
Comment (vdemeester) [inline integration-cli/docker_cli_tag_test.go]: I would have done something more like `c.Assert(err, check.NotNil, check.Commentf("tag busybox %v should have failed : %v", repo, out))`.

File: integration-cli/docker_cli_tag_test.go [hunk 8]
```diff
@@ -45,10 +43,8 @@ func (s *DockerSuite) TestTagInvalidPrefixedRepo(c *check.C) {
 	invalidTags := []string{"repo:fo$z$", "repo:Foo@3cc", "repo:Foo$3", "repo:Foo*3", "repo:Fo^3", "repo:Foo!3", "repo:%goodbye", "repo:#hashtagit", "repo:F)xcz(", "repo:-foo", "repo:..", longTag}
 
 	for _, repotag := range invalidTags {
-		_, _, err := dockerCmdWithError("tag", "busybox", repotag)
-		if err == nil {
-			c.Fatalf("tag busybox %v should have failed", repotag)
-		}
+		out, _, err := dockerCmdWithError("tag", "busybox", repotag)
+		c.Assert(err, check.NotNil, check.Commentf(out))
```
Comment (vdemeester) [inline integration-cli/docker_cli_tag_test.go]: Same here

**Comment by HuKeping** [comment]
@vdemeester updated.

**Comment by HuKeping** [comment]
ping @vdemeester @cpuguy83

**Comment by heartlock** [comment]
@HuKeping how to replace a standard comment with commentf when there are no vars being interpolated

**Comment by HuKeping** [comment]
You mean use
```
c.Assert(err, check.IsNil)  // this is the message
```

instead of 
```
c.Assert(err, check.IsNil, check.Commentf("this is the message")
```

I personally not like the first one.

**Comment by heartlock** [comment]
I mean 
c.Assert(i >= maxAttempts, checker.Equals, false, check.Commentf("timeout reached: image was not pulled after client disconnected"))
replace Commentf as there are no vars in() just like print and printf

**Comment by HuKeping** [comment]
I think you misunderstand what @MHBauer means, he is not saying use something like `check.Comment` but a regular Golang comment.

**Comment by heartlock** [comment]
like this
`c.Assert(i >= maxAttempts, checker.Equals, false))  //timeout reached: image was not pulled after client disconnected`
???
I don't like

**Comment by HuKeping** [comment]
correct, the `c.Assert` can use the context after "//" when there is error happen.

**Comment by heartlock** [comment]
@HuKeping thanks for your help :wink:

**Comment by HuKeping** [comment]
ping @vdemeester

**Comment by vdemeester** [comment]
It would be perfect to use `checker` for all the checker but that's a nit :wink:.

LGTM 🐰

File: integration-cli/docker_cli_top_test.go [hunk 9]
```diff
@@ -45,14 +37,9 @@ func (s *DockerSuite) TestTopPrivileged(c *check.C) {
...
-	if !strings.Contains(out1, "top") && !strings.Contains(out2, "top") {
-		c.Fatal("top should've listed `top` in the process list, but failed twice")
-	} else if !strings.Contains(out1, "top") {
-		c.Fatal("top should've listed `top` in the process list, but failed the first time")
-	} else if !strings.Contains(out2, "top") {
-		c.Fatal("top should've listed `top` in the process list, but failed the second itime")
-	}
+	dockerCmd(c, "kill", cleanedContainerID)
 
+	c.Assert(!strings.Contains(out1, "top") && !strings.Contains(out2, "top"), check.Equals, false, check.Commentf("top should've listed `top` in the process list, but failed twice"))
```
Comment (cpuguy83) [inline integration-cli/docker_cli_top_test.go]: Seems like we are checking this below and should remove this.

**Comment by HuKeping** [comment]
@vdemeester the checker has all the other `checkers` now?

**Comment by vdemeester** [comment]
yay perfect :wink:

**Comment by cpuguy83** [comment]
LGTM

**Pull request merged**